src/AzureCloudServiceBus/ServiceBusManager.cs

```diff
@@ -38,19 +38,55 @@ private async Task EnsureAuthenticatedAsync()
             }
         }
 
-        internal async Task<string> CreateNamespaceAsync(string location)
+        internal async Task<string> CreateRandomNamespaceAsync(string location)
         {
             await EnsureAuthenticatedAsync();
 
             var ns = $"squadron-{Guid.NewGuid().ToString("N").Substring(8)}";
+            return await CreateNamespaceAsync(location, ns);
+        }
+
+        internal async Task<string> CreateNamespaceIfNotExistsAsync(string location, string serviceBusNamespace)
+        {
+            await EnsureAuthenticatedAsync();
+
+            if (await CheckServiceBusNamespaceExists(serviceBusNamespace) == false)
+                return await CreateNamespaceAsync(location, serviceBusNamespace);
+
+            _identifier.Name = serviceBusNamespace;
+            return serviceBusNamespace;
+        }
+
+        private async Task<bool> CheckServiceBusNamespaceExists(string serviceBusNamespace)
+        {
+            var serviceBusNamespaces = new List<SBNamespace>();
```

Comment:
Insead of listing all the service bus namespaces I would use `_client.Namespaces.CheckNameAvailabilityMethodAsync`

Reply:
Good point, and we have been there as well. The reason why we didn't go with `_client.Namespaces.CheckNameAvailabilityMethodAsync` is that it will return the global availability of a name. In this context, we need to know the availability of a name given a resource group.

Reply:
ServiceBus namespace name must be global available because has a public endpoint.
https://docs.microsoft.com/en-us/rest/api/servicebus/create-namespace `...the name must be unique across Azure to be successfully created`

Reply:
@glucaci did you see my comment [here](https://github.com/SwissLife-OSS/squadron/pull/95#issuecomment-916674298)?